src/videojs-hls.js

```diff
@@ -854,7 +854,7 @@ videojs.HlsHandler.prototype.fillBuffer = function(mediaIndex) {
   // we have entered a state where we are fetching the same segment,
   // try to walk forward
   if (this.lastSegmentLoaded_ &&
-      this.lastSegmentLoaded_ === this.playlistUriToUrl(segment.uri)) {
+      this.lastSegmentLoaded_ === segment) {
```

Comment:
When we have to reload the playlist because the video is live, I believe we recreate the segment objects and merge the properties together, which would cause this check to fail.